blinds/blinds_schedule.py

```diff
@@ -330,6 +337,49 @@ def hasConflict( timeBlockList ):
 
         return False
 
+    '''
+    Parses a timezone formatted like GMT-0600 and returns a pytz timezone for it. 
+    Currently does not support half hour differences from GMT, as these are not included in the 
+    list of supported timezones in pytz for Etc/GMT-X. While they do exist for the specific place names, the conversion 
+    is much more difficult.
+    '''
+    @staticmethod
+    def tzFromGmtString( gmtStr ):
+        tzRegex = r"GMT([+-])(\d{4})" # matches things like GMT-0600
+        gmtTimezoneBase = "Etc/GMT"
+        
+        matches = re.search( tzRegex, gmtStr)
+
+        if not matches:
+            raise InvalidTimeZoneStringException( "Invalid timezone string: " + gmtStr )
+
+        plusminus = matches.group(1)
+        offset = int( matches.group(2) ) / 100
+
+        if ( offset != int( offset ) or offset > 12 or offset < -14 ):
+            raise InvalidTimeZoneStringException( "Offset must be an integer number of hours between -14 and 12" )
+        
+        return timezone( gmtTimezoneBase + plusminus + str( int( offset ) ) )
+
+    '''
+    Converts a pytz timezone of the type Etc/GMT-X to a string formatted like "GMT-XXXX" 
+    '''
+    @staticmethod
+    def tzToGmtString( tz ):
+        tzNameRegex = r"Etc/GMT([+-])(\d+)"
+        matches = re.search( tzNameRegex, tz.zone)
```

Comment:
You forgot null check for matches, it causes your tests to fail.

Reply:
Fixed now.